submissions/models.py

```diff
@@ -124,6 +123,25 @@ class Submission(models.Model):
     # name so it continues to use `raw_answer`.
     answer = JSONField(blank=True, db_column="raw_answer")
 
+    # Has this submission been soft-deleted? This allows instructors to reset student
+    # state on an item, while preserving the previous value for potential analytics use.
+    DELETED = 'D'
+    ACTIVE = 'A'  # Note that, for migration purposes, a status value of None will be handled as 'Active'.
+    # If you are adding more statuses, be *sure* to write a data migration first to handle this!
+    STATUS_CHOICES = (
+        (DELETED, 'Deleted'),
+        (ACTIVE, 'Active'),
+    )
+    status = models.CharField(max_length=1, choices=STATUS_CHOICES, null=True, default=None)
+
+    # Override the default Manager with our custom one to filter out soft-deleted items
+    class SoftDeletedManager(models.Manager):
```

Comment:
Maybe rename to something like `ExcludeDeletedManager`?  Not sure about the name, but the current name sounds like it would provide the DELETED records.